**적합도 검정**
- 사례) 한 학교에서 4개의 전공(A,B,C,D) 선택 비율이 동일하다고 주장 → 실제 조사 데이터와 비교


In [1]:
obs = [40, 50, 45, 65]

1. 통계적 가설 검정
 - 귀무가설(H0) : 전공별 학생 비율은 모두 동일하다.
 - 대립가설(H1) : 전공별 학생 비율은 동일하지 않다.
2. 유의수준 결정
 - 유의수준 𝜶=𝟎.𝟎𝟓

3. 적합도 검정

In [2]:
from scipy import stats

chi2_stat, p_val = stats.chisquare(obs)
print("검정통계량:", chi2_stat)
print("p-value:", p_val)

검정통계량: 7.0
p-value: 0.07189777249646509


4. 결과도출
 - chi2_statistic과 p-value를 확인하고, p-value < 0.05이면 귀무가설 기각
 - 해석 : 유의수준 0.05에서 p-value가 0.0719이므로, 귀무가설을 채택 → 전공별 비율이 동일하다.



※ 전공별 학생 수가 균등하게 분포하지 않을 것으로 예상된다면, stats.chisquare(obs, exp)처럼 기대빈도(exp)를 직접 지정하여 분석을 수행
 - 해석 : 귀무가설을 기각 → 전공별 비율이 기대한 [60, 60, 40, 40]과는 동일하지 않다.

In [5]:
exp = [60, 60, 40, 40]
chi2_stat, p_val = stats.chisquare(obs, exp)
print("검정통계량:", chi2_stat)
print("p-value:", p_val)

검정통계량: 24.583333333333336
p-value: 1.8869203817365673e-05


---
**독립성 검정**
- 사례) 성별과 흡연 여부가 독립적인지 조사



In [ ]:
import pandas as pd

data = pd.DataFrame({
    '흡연':    [40, 20],
    '비흡연':  [60, 80]
}, index=['남', '여'])

data

,흡연,비흡연
남,40,60
여,20,80


1. 통계적 가설 검정
 - 귀무가설(H0) : 성별과 흡연 여부는 서로 독립이다.
 - 대립가설(H1) : 성별과 흡연 여부는 서로 독립이 아니다.
2. 유의수준 결정
 - 유의수준 𝜶=𝟎.𝟎𝟓

3. 독립성 검정

In [ ]:
chi2, p, dof, exp = stats.chi2_contingency(data)
print("검정통계량:", chi2)
print("p-value:", p)
print("자유도:", dof)
print("기대도수:\n", pd.DataFrame(exp, index=data.index, columns=data.columns))

검정통계량: 8.595238095238095
p-value: 0.0033704314473977145
자유도: 1
기대도수:
      흡연   비흡연
남  30.0  70.0
여  30.0  70.0


4. 결과도출
 - chi2_statistic과 p-value, 자유도, 기대빈도를 확인하고, p-value < 0.05이면 귀무가설 기각

 -  해석 : 유의수준 0.05에서 p-value가 0.003이므로, 귀무가설을 기각 → 성별과 흡연 여부는 독립이 아니다. (관련이 있다.)


---
**동질성 검정**
- 사례) 세 개 도시(서울, 부산, 대구)에서 교통수단 이용 비율이 동일한지 조사


In [ ]:
df = pd.DataFrame({
    "도시": [
        "서울"] * 10 + ["부산"] * 10 + ["대구"] * 10,
    "교통수단": [
        "버스", "버스", "버스", "지하철", "지하철", "지하철", "택시", "택시", "택시", "택시",
        "버스", "버스", "지하철", "지하철", "지하철", "택시", "택시", "택시", "택시", "택시",
        "버스", "버스", "버스", "버스", "지하철", "지하철", "지하철", "택시", "택시", "택시"
    ]
})

1. 통계적 가설 검정
 - 귀무가설(H0) : 세 도시의 교통수단 이용 비율은 동일하다.
 - 대립가설(H1) : 세 도시의 교통수단 이용 비율은 동일하지 않다
2. 유의수준 결정
 - 유의수준 𝜶=𝟎.𝟎𝟓

※ 교차표로 변경

In [ ]:
table = pd.crosstab(df["도시"], df["교통수단"])
table

교통수단,버스,지하철,택시
도시,,,
대구,4,3,3
부산,2,3,5
서울,3,3,4


3. 동질성 검정

In [ ]:
chi2, p, dof, expected = stats.chi2_contingency(table)

print("검정통계량:", chi2)
print("p-value:", p)
print("자유도:", dof)
print("\n기대도수:\n", pd.DataFrame(expected, index=table.index, columns=table.columns))

검정통계량: 1.1666666666666665
p-value: 0.8835556474692412
자유도: 4

기대도수:
 교통수단   버스  지하철   택시
도시                 
대구    3.0  3.0  4.0
부산    3.0  3.0  4.0
서울    3.0  3.0  4.0


4. 결과도출
 - chi2_statistic과 p-value, 자유도, 기대빈도를 확인하고, p-value < 0.05이면 귀무가설 기각
 - 해석 : 유의수준 0.05에서 p-value가 0.8835이므로, 귀무가설을 채택 → 세 도시의 교통수단 이용 비율이 동일하다.
